updating Privacy Considerations (#226)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

# Privacy Considerations {#privacy_considerations}

> The privacy principles of [@ISO.29100] MUST be adhered to.

Reviewer: ```suggestion
The privacy principles of [@ISO.29100] should be adhered to.
```

> The privacy principles of [@ISO.29100] MUST be adhered to. 

Reviewer: I don't feel it's appropriate to have a normative MUST pointing to a document that isn't freely available.

Author: It is actually publicly available - see url and search for the number

Reviewer: My mistake. Sorry. I guess I'm accustomed to ISO docs being difficult to access.  Even so, I don't think a normative MUST to a different document is okay.

Author: I agree must is too strong. Would should be acceptable?

```suggestion
The privacy principles of [@ISO.29100] should be adhered to. 
```

Reviewer: little should would be better, yes.

Reviewer: github is telling me this conversation is outdated though so I can't commit the suggestion

> The privacy principles of [@ISO.29100] MUST be adhered to. 

Author: ```suggestion
The privacy principles of [@ISO.29100] should be adhered to. 
```

> The privacy principles of [@ISO.29100] should be adhered to.

Reviewer 2: Intentional lower-case should?

Reviewer 2: Just saw the conversation pertaining to this.

mechanism.

> Implementers MUST ensure that the transport protocol provides confidentiality,

Reviewer 2: ```suggestion
Implementers MUST ensure that the transport protocol provides confidentiality
```

> if the privacy of End-User data or correlation attacks by the passive observers are a concern. Implementers MAY define an

Reviewer 2: ```suggestion
if the privacy of End-User data or correlation attacks by passive observers are a concern. Implementers MAY define an
```

- Use advanced cryptographic schemes, outside the scope of this specification.
- Issue a batch of SD-JWTs to the Holder to enable the Holder to use a unique SD-JWT per Verifier. This only helps with Verifier/Verifier unlinkability.

> Colluding Verifier/Verifier pairs correspond to RP+RP'-U Unlinkability as defined in [@ISO.27551].

Reviewer: I don't have access to ISO.27551 and am not going to pay for it. I doubt I'm unique there.  `RP+RP'-U` looks odd but what do I know? What value does this sentence bring?

Author: no, you are probably not unique. 
one value might be helping readers familiar with ISO 27551 draw the parallel between SD-JWT spec and ISO text, which is not an absolute requirement and I have not seen other IETF specs really do it.

Reviewer: I'd suggest the sentence be dropped.

Reviewer: ```suggestion
```

Reviewer: ```suggestion
```

Author: ```suggestion
```

## Issuer Issuing One Type of SD-JWT

Issuer issuing only one type of SD-JWT might have privacy implications, because if the Holder has an SD-JWT issued by that Issuer, its type and claim names can be determined.

For example, if the National Cancer Institute only issued SD-JWTs with cancer registry information, it is possible to deduce that the Holder owning its SD-JWT is a cancer patient.

> To mitigate this issue, a group of issuers may elect to use a common Issuer identifier, or use a group signature scheme instead of an individual signature.

Reviewer: There's not any group signatures registered or spec'd for JWS/JWT (that I know of anyway) so that's not really a meaningful suggestion.

Author: agreed. though, we do say in section 9.4 `Use advanced cryptographic schemes, outside the scope of this specification`.

```suggestion
To mitigate this issue, a group of issuers may elect to use a common Issuer identifier. A group signature scheme outside the scope of this specification may also be used, instead of an individual signature.
```

- Use advanced cryptographic schemes, outside the scope of this specification.
- Issue a batch of SD-JWTs to the Holder to enable the Holder to use a unique SD-JWT per Verifier. This only helps with Verifier/Verifier unlinkability.

## Issuer Identifier

> Issuer issuing only one type of SD-JWT might have privacy implications, because if the Holder has an SD-JWT issued by that Issuer, its type and claim names can be determined.
> 
> For example, if the National Cancer Institute only issued SD-JWTs with cancer registry information, it is possible to deduce that the Holder owning its SD-JWT is a cancer patient.
> 
> Moreover, issuer identifier alone may reveal information about the user.
> 
> For example, when a military organization or a drug rehab center issues a vaccine credential, verifiers can deduce that the holder is a military member or has a substance use disorder.

Reviewer 2: ```suggestion
An Issuer issuing only one type of SD-JWT might have privacy implications, because if the Holder has an SD-JWT issued by that Issuer, its type and claim names can be determined.

For example, if the National Cancer Institute only issued SD-JWTs with cancer registry information, it is possible to deduce that the Holder owning its SD-JWT is a cancer patient.

Moreover, the issuer identifier alone may reveal information about the user.

For example, when a military organization or a drug rehabilitation center issues a vaccine credential, verifiers can deduce that the holder is a military member or may have a substance use disorder.
```

Reviewer 2: Good examples!

- Use advanced cryptographic schemes, outside the scope of this specification.
- Issue a batch of SD-JWTs to the Holder to enable the Holder to use a unique SD-JWT per Verifier. This only helps with Verifier/Verifier unlinkability.

Colluding Verifier/Verifier pairs correspond to RP+RP'-U Unlinkability as defined in [@ISO.27551].

## Issuer Identifier

Issuer issuing only one type of SD-JWT might have privacy implications, because if the Holder has an SD-JWT issued by that Issuer, its type and claim names can be determined.

For example, if the National Cancer Institute only issued SD-JWTs with cancer registry information, it is possible to deduce that the Holder owning its SD-JWT is a cancer patient.

> Moreover, issuer identifier alone may reveal information about the user. 

Reviewer: ```suggestion
Moreover, issuer identifier alone may reveal information about the user.
```

remove the trailing whitespace

> For example, when a military organization or a drug rehab center issues a vaccine credential, verifiers can deduce that the holder is a military member or has a substance use disorder.

Author: updated the text because confirmed with the smart health community for below use-case:
> A scenario we have seen is something like a military organization or a rehab clinic issuing a vaccine credential; if the iss is directly associated with one of theses orgs, it allow verifiers to deduce that the holder is a military member or has a substance use disorder. That's undesirable, and indeed we've seen issuers adopt generic names or maybe even (I'm not sure about this) join up with other issuing orgs to pool populations.

Reviewer: makes sense

  </front>
</reference>

<reference anchor="ISO.29100" target="https://standards.iso.org/ittf/PubliclyAvailableStandards/index.html">
  <front>
    <title>ISO/IEC 29100:2011 Information technology — Security techniques — Privacy framework</title>
   <date year="2011"/>
  </front>
</reference>

> <reference anchor="ISO.27551" target="https://www.iso.org/standard/72018.html">
>   <front>
>     <title>ISO/IEC 27551:2021 Information security, cybersecurity and privacy protection — Requirements for attribute-based unlinkable entity authentication</title>
>    <date year="2011"/>
>   </front>
> </reference>
> 

Author: ```suggestion
```
